src/Services/ConnectionManager.vala

```diff
@@ -157,25 +157,30 @@ public class Sequeler.Services.ConnectionManager : Object {
 
 		Quark q = Quark.from_string ("ssh-error-str");
 		var home_dir = Environment.get_home_dir ();
-		var keyfile1 = home_dir + "/.ssh/id_rsa.pub";
-		var keyfile2 = home_dir + "/.ssh/id_rsa";
+
+		// private key file
+		var keyfile2 = GLib.Filename.from_uri(data["ssh_identity_file"]);
+
+		// public key file
+		var keyfile1 = keyfile2 + ".pub";
+
 
 		// SSH credentials if password authentication is required
 		var username = data["ssh_username"];
 		var password = data["ssh_password"];
 
 		// SSH HOST address and Port
 		var server_ip = data["ssh_host"];
-		var server_port = data["ssh_port"] != "" ? (uint16) (data["ssh_port"]).hash () : 22;
+		var server_port = data["ssh_port"] != "" ? (uint16) int.parse (data["ssh_port"]) : 22;
 
 		// The IP address where the DB is available on your SSH
-		var local_listenip = data["host"] != "" ? data["host"] : "127.0.0.1";
+		var local_listenip = "127.0.0.1";
 
 		// The Port used by the DB on your SSH host
 		uint16 local_listenport = 9000;
 
 		// Default vars for TCPIP Tunnelling
-		var remote_desthost = "127.0.0.1";
+		var remote_desthost = data["host"] != "" ? data["host"] : "127.0.0.1";
```

Comment:
I'm not sure about this.
Usually your remote host when connecting via SSH is always local, as your tunnel takes care of that.
Why this change?

Reply:
In my particular use case I connect via SSH to a host that is not my database server, but it has a VPN connection that I can use to reach my database.

Thus, a database connection to `127.0.0.1:9000` is tunneled through` data["ssh_host"]:data["ssh_port"]` then routed to `data["host"]:data["port"]`

I'm aware it's not so common, but I think this approach is more flexible as you can retain old behavior by leaving `data["host"]` and `data["port"]` blank, while setting them to something else in case the database server host is different from the SSH server host.